src/server/requestHandlers.js

```diff
@@ -222,45 +193,85 @@ sjrk.storyTelling.server.saveStoryToDatabase = function (dataSource, binaryRenam
     });
 };
 
-/**
- * Update any media URLs to refer to the changed file names
- *
- * @param {Object} blocks - a collection of story blocks with file references
- * @param {Object} files - a list of files to refer to
+// Kettle request handler for saving a single file associated with a pre-existing story
+fluid.defaults("sjrk.storyTelling.server.saveStoryFileHandler", {
+    gradeNames: "kettle.request.http",
+    requestMiddleware: {
+        saveStoryFile: {
+            middleware: "{server}.saveStoryFile"
+        },
+        removePreviousStoryFile: {
+            middleware: "{server}.removePreviousStoryFile",
+            priority: "after:saveStoryFile"
+        }
+    },
+    invokers: {
+        handleRequest: {
+            funcName: "sjrk.storyTelling.server.handleSaveStoryFile",
+            args: ["{arguments}.0", "{server}.storyDataSource", "{server}.options.globalConfig.authoringEnabled"]
+        }
+    }
+});
+
+/**
+ * Saves a single file to the server filesystem and responds to the HTTP request.
+ * If the file is an image,
+ * Errors will be raised in the following situations:
+ * - Authoring is not enabled
+ * - An error is encountered while trying to get the story (invalid ID, DB offline, etc.)
+ * - The story that the file is associated with is already published
+ * - The provided file is not valid
+ * - An error is encountered while rotating an image to its correct orientation
  *
- * @return {Object.<String, String>} - a key-value collection linking uploaded filenames to server paths
+ * @param {Object} request - a Kettle request containing a single file associated with a story
+ * @param {Component} dataSource - an instance of sjrk.storyTelling.server.dataSource.couch.story
+ * @param {Boolean} authoringEnabled - a server-level flag to indicate whether authoring is enabled
  */
-sjrk.storyTelling.server.buildBinaryRenameMap = function (blocks, files) {
-    // key-value pairs of original filename : generated filename
-    // this is used primarily by tests, but may be of use
-    // to client-side components too
-    var binaryRenameMap = {};
-
-    fluid.each(blocks, function (block) {
-        if (block.blockType === "image" || block.blockType === "audio" || block.blockType === "video") {
-            if (block.fileDetails) {
-                // Look for the uploaded file matching this block
-                var mediaFile = fluid.find_if(files, function (file) {
-                    return file.originalname === block.fileDetails.name;
-                });
-
-                // If we find a match, update the media URL. If not, clear it.
-                if (mediaFile) {
-                    sjrk.storyTelling.server.setMediaBlockUrl(block, mediaFile.filename);
-                    binaryRenameMap[mediaFile.originalname] = mediaFile.filename;
+sjrk.storyTelling.server.handleSaveStoryFile = function (request, dataSource, authoringEnabled) {

```

Comment:
I'm not sure I'm following the workflow of this function. It says it handles saving, but I'm not exactly sure where that happens. It seems that it is already saved by the time this handler executes and that this function will attempt to rotate the image as needed and also verifies the save. Is that correct?